testscript/cmd.go

```diff
@@ -111,6 +111,11 @@ func (ts *TestScript) cmdCmpenv(neg bool, args []string) {
 
 func (ts *TestScript) doCmdCmp(neg bool, args []string, env bool) {
 	name1, name2 := args[0], args[1]
+	if name1 == name2 {
+		// It's surprisingly easy to do e.g. `cmp foo.go foo.go` rather than
+		// `cmp foo.go foo.go.want`; protect against such obvious cases.
+		ts.Fatalf("cmp: comparing the same file twice")
```

Comment:
It's really only being compared once. How about "cannot compare a file against itself" ?

Reply:
ah, I meant "twice" here as in "shows up twice in the argument list", but I agree it's ambiguous. done.